schedview/collect/nightreport.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+from typing import Literal
+
+import requests
+
+from schedview.dayobs import DayObs
+
+MAX_RETRIES = 1
+
+EXCLUDED_COMPONENTS_FOR_TELESCOPE = {
+    "AuxTel": ["MTMount", "MainTel"],
+    "Simonyi": ["AuxTel", "ATMCS", "ATDome"],
+}
+
+
+def _get_with_retries(api_endpoint, params):
+    response = requests.get(api_endpoint, params)
+    try_number = 1
+    while not response.status_code == 200:
+        if try_number > MAX_RETRIES:
+            response.raise_for_status()
+        response = requests.get(api_endpoint, params)
+        try_number += 1
+
+    return response.json()
+
+
+def get_night_report(
+    day_obs: DayObs | str | int,
+    telescope: Literal["AuxTel", "Simonyi"],
+    api_endpoint: str = "https://usdf-rsp-dev.slac.stanford.edu/nightreport/reports",
```

Comment:
We need a better way to handle these endpoints. 
Presumably these are tools that should run at summit or at USDF - these services are available in both. 
The base URL differs, but the endpoint "nightreport/reports" is the same.
I don't have a super awesome suggestion at present, but one option might be something based on:
```
def get_clients() -> dict:
    """Return site-specific client connections. 

    Returns
    -------
    endpoints : `dict`
        Dictionary with `efd`, `obsenv`, 
        `narrative_log`, and `exposure_log`
        connection information.
        For the obsenv, narrative log and exposure log, these are only
        defined for the summit or USDF.
    """
    site = getSite()
    # This will fail if site is "unknown"
    # although could go with usdf_efd ..
    efd_client = makeEfdClient()
    if site == "summit":
        API_BASE = "https://summit-lsp.lsst.codes/"
        obsenv_client = EfdClient('summit_efd', db_name='lsst.obsenv')
    elif site == "tucson" or site == "base":
        # I don't know what the logging looks like here, 
        # but suspect it doesn't exist
        obsenv_client = None
        API_BASE = None
    else:
        obsenv_client = EfdClient('usdf_efd', db_name='lsst.obsenv')
        API_BASE = "https://usdf-rsp.slac.stanford.edu/"
    narrative_log_url =  API_BASE + "narrativelog/messages"
    exposure_log_url = API_BASE + "exposurelog/messages"
    return {'efd': efd_client, 'obsenv': obsenv_client, 
            'narrative_log': narrative_log_url, 'exposure_log': exposure_log_url}
```
although I think in practice, we don't want to use summit_utils' makeEfdClient tool for getting the EfdClient, for example (since it fails outside of any of the known places) and if we drop that, we may want to just drop getSite too.

Reply:
We could figure out where it's running with something like `'.'.join(socket.getfqdn().split('.')[1:])` and have a dictionary somewhere that maps domains to base URLs, but maybe that should be a different PR?
What prompted me to make this PR now, even though it is a bit rough, is that it's getting to large, with too big a variety of stuff in it for one PR. I'd rather merge what I have so far and incrementally improve it than develop this onto one big monster PR.